integration-tests/utils/integration-helpers.js

```diff
@@ -7,47 +7,45 @@ const { withUserToken } = require('../utils/terra-sa-utils')
 
 const defaultTimeout = 5 * 60 * 1000
 
-const makeWorkspace = async ({ billingProject, context, testUrl, token }) => {
-  const ajaxPage = await context.newPage()
-
-  await ajaxPage.goto(testUrl)
-  await signIntoTerra(ajaxPage, token)
+const withSignedInPage = fn => async options => {
+  const { context, testUrl, token } = options
+  const page = await context.newPage()
+  try {
+    await page.goto(testUrl)
+    await signIntoTerra(page, token)
+    return await fn({ ...options, page })
+  } finally {
+    await page.close()
+  }
+}
 
+const makeWorkspace = withSignedInPage(async ({ page, billingProject }) => {
   const workspaceName = `test-workspace-${Math.floor(Math.random() * 100000)}`
 
-  await ajaxPage.evaluate((name, billingProject) => {
+  await page.evaluate((name, billingProject) => {
     return window.Ajax().Workspaces.create({ namespace: billingProject, name, attributes: {} })
   }, workspaceName, billingProject)
 
   console.info(`created workspace: ${workspaceName}`)
 
-  await ajaxPage.close()
-
   return workspaceName
-}
-
-const deleteWorkspace = async (workspaceName, { billingProject, context, testUrl, token }) => {
-  const ajaxPage = await context.newPage()
-
-  await ajaxPage.goto(testUrl)
-  await signIntoTerra(ajaxPage, token)
+})
 
-  await ajaxPage.evaluate((name, billingProject) => {
+const deleteWorkspace = withSignedInPage(async ({ page, billingProject, workspaceName }) => {
+  await page.evaluate((name, billingProject) => {
     return window.Ajax().Workspaces.workspace(billingProject, name).delete()
   }, workspaceName, billingProject)
 
   console.info(`deleted workspace: ${workspaceName}`)
-
-  await ajaxPage.close()
-}
+})
 
 const withWorkspace = test => async options => {
   const workspaceName = await makeWorkspace(options)
 
   try {
     await test({ ...options, workspaceName })
   } finally {
-    await deleteWorkspace(workspaceName, options)
+    await deleteWorkspace({ ...options, workspaceName })
```

Comment:
Had to think about this for a second, but it makes sense to me now. Particularly important that it's this and not `{ workspaceName, ...options }`.